File: android-feature/show-details/src/main/kotlin/com/thomaskioko/tvmaniac/showdetails/ui/components/SeasonProgressCard.kt
package com.thomaskioko.tvmaniac.showdetails.ui.components

import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Surface
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.thomaskioko.tvmaniac.compose.components.ThemePreviews
import com.thomaskioko.tvmaniac.compose.components.TvManiacChip
import com.thomaskioko.tvmaniac.compose.theme.TvManiacTheme
import com.thomaskioko.tvmaniac.presenter.showdetails.model.SeasonModel

@Composable
internal fun SeasonProgressCard(
    season: SeasonModel,
    isSelected: Boolean,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    TvManiacChip(
        modifier = modifier,
        text = season.name,
        selected = isSelected,
        onClick = onClick,
    )
}

@ThemePreviews
@Composable
private fun SeasonProgressCardPreview() {
    TvManiacTheme {
        Surface {
            SeasonProgressCard(
                season = SeasonModel(
                    seasonId = 1L,
                    tvShowId = 1L,
                    name = "Season 1",
                    seasonNumber = 1L,
                    watchedCount = 4,
                    totalCount = 6,
                ),
                isSelected = false,
                onClick = {},
                modifier = Modifier.padding(4.dp),
            )
        }
    }
}

@ThemePreviews
@Composable
private fun SeasonProgressCardSelectedPreview() {
    TvManiacTheme {
        Surface {
            SeasonProgressCard(
                season = SeasonModel(
                    seasonId = 1L,
                    tvShowId = 1L,
                    name = "Season 1",
                    seasonNumber = 1L,
                    watchedCount = 6,
                    totalCount = 6,
                ),
                isSelected = true,
                onClick = {},
                modifier = Modifier.padding(4.dp),
            )
        }
    }
}

File: ios/Modules/SwiftUIComponents/Sources/SwiftUIComponents/Components/SeasonProgress/SeasonProgressCard.swift
import SwiftUI

public struct SeasonProgressCard: View {
    @Theme private var theme

    private let season: SwiftSeason
    private let isSelected: Bool
    private let onClick: () -> Void

    public init(
        season: SwiftSeason,
        isSelected: Bool,
        onClick: @escaping () -> Void
    ) {
        self.season = season
        self.isSelected = isSelected
        self.onClick = onClick
    }

    public var body: some View {
        ChipView(
            label: season.name,
            isSelected: isSelected,
            action: onClick
        )
    }
}

#Preview {
    VStack(spacing: 16) {
        SeasonProgressCard(
            season: SwiftSeason(
                tvShowId: 1,
                seasonId: 1,
                seasonNumber: 1,
                name: "Season 1",
                watchedCount: 4,
                totalCount: 6,
                progressPercentage: 0.67
            ),
            isSelected: false,
            onClick: {}
        )

        SeasonProgressCard(
            season: SwiftSeason(
                tvShowId: 1,
                seasonId: 2,
                seasonNumber: 2,
                name: "Season 2",
                watchedCount: 6,
                totalCount: 6,
                progressPercentage: 1.0
            ),
            isSelected: true,
            onClick: {}
        )
    }
    .padding()
    .environment(\.tvManiacTheme, DarkTheme())
}
